ui/widgets/nuxeo-collapsible.js

```diff
@@ -0,0 +1,155 @@
+/**
+@license
+(C) Copyright Nuxeo Corp. (http://nuxeo.com/)
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import '@polymer/iron-collapse/iron-collapse.js';
+
+import '@polymer/iron-flex-layout/iron-flex-layout.js';
+import '@polymer/iron-icon/iron-icon.js';
+import '@polymer/iron-icons/hardware-icons.js';
+import '@nuxeo/nuxeo-elements/nuxeo-element.js';
+import '@polymer/polymer/lib/elements/dom-if.js';
+import { html } from '@polymer/polymer/lib/utils/html-tag.js';
+
+{
+  /**
+   * An element for creating a collapsible block of content.
+   * It comes with a default chevron that can be displayed at the end (default) or at the beggining of the heading.
+   * The heading can be expanded by default.
+   * The chevron has its own slot so it can be also customized.
+   * It is also possible to disable the default animation.
+   *
+   * Example:
+   *
+   *     <nuxeo-collapsible expanded chevronPosition="start">
+   *        <label slot="heading">Nuxeo</label>
+   *        <div>Collapsed content</div>
+   *     </nuxeo-collapsible>
+   *
+   * ### Styling
+   * The following custom properties and mixins are available for styling:
+   * Custom property | Description | Default
+   * ----------------|-------------|----------
+   * `--nuxeo-collapsible-heading-padding` | Heading padding | `4px`
+   * `--nuxeo-collapsible-icon-height` | Chevron height | `20px`
+   * `--nuxeo-collapsible-icon-width` | Chevron width | `20px`
+   *
+   * @memberof Nuxeo
+   *
+   */
+
+  class Collapsible extends Nuxeo.Element {
+    static get template() {
+      return html`
+        <style>
+          .toggle {
+            --iron-icon-height: var(--nuxeo-collapsible-icon-height, 20px);
+            --iron-icon-width: var(--nuxeo-collapsible-icon-width, 20px);
+          }
+
+          button {
+            height: 100%;
```

Comment:
because we're using a button, on focus there's a blue target border we don't see anywhere else on nuxeo elements, not sure if we can override it.

Reply:
While developing this element I had in mind all the time accessibility concerns. Having that default focus border it's fundamental for achieving a decent experience for all the users and we should promote its usage. After saying this, we may need to rethink this state throughout our elements @carolinagouveia

Reply:
Yes agree, this state will also be added to other elements for accessibility purposes, it's something we're currently missing.